### PR TITLE
Setup composer lint

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

**Comment by SenseException** [comment]
Does it make sense to add a new workflow to doctrine/.github?

File: .github/workflows/composer-lint.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  push:
+    branches:
+      - "*.x"
+    paths:
+      - "composer.json"
+
+jobs:
+  composer-lint:
+    name: "Composer Lint"
+    uses: "doctrine/.github/.github/workflows/composer-lint.yml@2.1.0"
```
Comment (greg0ire) [inline .github/workflows/composer-lint.yml:18]: @SenseException It does, so much that we did, and it's included here :wink:

File: .github/workflows/composer-lint.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    with:
+      php-version: "8.1"
```
Comment (derrabus) [inline .github/workflows/composer-lint.yml]:
```suggestion
```

Isn't that the default?

File: .github/workflows/composer-lint.yml (hunk 1 above)
Comment (SenseException) [reply .github/workflows/composer-lint.yml:18]: Uhm, I need to clean my glasses...

Comment (derrabus) [reply .github/workflows/composer-lint.yml:18]: Either I'm getting senile or I've had a beer too much already or an earlier version of this PR pointed to @greg0ire's fork of `.github` instead. 🤔

File: .github/workflows/composer-lint.yml (hunk 2 above)
Comment (greg0ire) [reply .github/workflows/composer-lint.yml]: Oh right, I copy/pasted from another repo.

**Review by SenseException — APPROVED** [review]

**Pull request merged**